src/server/debug/server/server.go

```diff
@@ -461,7 +462,7 @@ func (s *debugServer) collectCommits(tw *tar.Writer, pachClient *client.APIClien
 		},
 	}
 	if err := collectDebugFile(tw, "commits", "json", func(w io.Writer) error {
-		ctx, cancel := context.WithCancel(pachClient.Ctx())
+		ctx, cancel := context.WithCancel(rctx)
```

Comment:
It’s a bit surprising to make the context cancellable this deep in the call stack, especially when it doesn’t appear to be spawning any background tasks which might need cancellation.

I think that this could just be the enclosing context. Could either close over it, or add to the signature of `collectDebugFile` and its callback.

Reply:
It depends on what happens when ForEachCommit returns early with an error; I don't see it shutting down the stream and I think that's what this is trying to achieve.

Reply:
> I didn't write this code, and it works, so I'm not too interested in changing it here.

That’s fair!